packages/client/hmi-client/src/components/workflow/util.ts

```diff
@@ -134,3 +139,18 @@ export const getGraphDataFromDatasetCSV = (
 
 	return graphData;
 };
+
+export function getActiveOutput<S>(node: WorkflowNode<S>) {
+	return node.outputs.find((output) => output.id === node.active);
+}
+
+export type VariableMappings = CalibrateMap[] | CalibrateEnsembleMappingRow[] | SimulateEnsembleMappingRow[];
+
+export const modelVarToDatasetVar = (mapping: VariableMappings, modelVariable: string) => {
+	if (_.isEmpty(mapping) || isSimulateEnsembleMappingRow(mapping[0])) return '';
+	const found = (mapping as CalibrateMap[] | CalibrateEnsembleMappingRow[]).find((d) =>
+		isCalibrateMap(d) ? d.modelVariable === modelVariable : d.newName === modelVariable
+	);
+	if (!found) return '';
+	return isCalibrateMap(found) ? found.datasetVariable : found.datasetMapping;
+};
```

Comment:
What is this function `modelVarToDatasetVar` suppose to do? It seems to touch upon all types of operators?

Would it be better if the signature is like?
```
modelVarToDatasetVar = (mapping: VariableMappings, modelVariable: string, operatorType: string) => {
  if (operatorType === "calibrate") {
  } else if (operatorType === "simulate") {
   ...
  }
}
```

Reply:
This function grabs a datasets variable for a given model variable from the provided mapping. I think we can just branch based on the mapping type since each operator (that uses mapping) uses different mapping object. It only touches the operator that has mapping object (calibrate, calibrate & simulate ensemble).  I'll just update the code so that it's more clear to read.

Reply:
Done